core/src/main/java/com/amplifyframework/predictions/models/Table.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.models;
+
+import androidx.annotation.NonNull;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A representation of 2-D table in given dimensions.
+ * A table can be detected from an image of a document
+ * and be organized into this data type.
+ */
+public final class Table {
```

Comment:
Since your `Cell` class is tightly related to this `Table`, it could make sense to make the `Cell` as a static inner class of the Table. That way, they're related by design and the reader doesn't need to go find one or the other.

Reply:
will do